lib/Doctrine/ORM/Persisters/Collection/OneToManyPersister.php

```diff
@@ -166,24 +166,30 @@ public function loadCriteria(PersistentCollection $collection, Criteria $criteri
     }
 
     /** @throws DBALException */
-    private function deleteEntityCollection(PersistentCollection $collection): int
+    private function deleteEntityCollection(PersistentCollection $collection)
     {
         $mapping     = $collection->getMapping();
         $identifier  = $this->uow->getEntityIdentifier($collection->getOwner());
         $sourceClass = $this->em->getClassMetadata($mapping['sourceEntity']);
         $targetClass = $this->em->getClassMetadata($mapping['targetEntity']);
         $columns     = [];
         $parameters  = [];
+        $types       = [];
 
         foreach ($targetClass->associationMappings[$mapping['mappedBy']]['joinColumns'] as $joinColumn) {
             $columns[]    = $this->quoteStrategy->getJoinColumnName($joinColumn, $targetClass, $this->platform);
             $parameters[] = $identifier[$sourceClass->getFieldForColumn($joinColumn['referencedColumnName'])];
+            $types[]      = PersisterHelper::getTypeOfColumn($joinColumn['referencedColumnName'], $sourceClass, $this->em);
         }
 
         $statement = 'DELETE FROM ' . $this->quoteStrategy->getTableName($targetClass, $this->platform)
             . ' WHERE ' . implode(' = ? AND ', $columns) . ' = ?';
 
-        return $this->conn->executeStatement($statement, $parameters);
+        $numAffected = $this->conn->executeStatement($statement, $parameters, $types);
+
+        assert(is_int($numAffected) || is_string($numAffected));
```

Comment:
I meant this:

```suggestion
        assert(is_int($numAffected));
```

In that case, you will have to ignore an error about the possibility of `$numAffected` being a string, in case it's gigantic I guess.